runestone/poll/js/poll.js

```diff
@@ -212,6 +213,7 @@ export default class Poll extends RunestoneBase {
             }
             $(this.resultsDiv).append(list);
         }
+        this.indicate_component_ready();
```

Comment:
If its called in the base class why call it again here??  Shouldn't there be one and only one indication of when the component is ready.  Otherwise we may get multiple firings of things that should not be done multiple times.

Reply:
It's only called in the base class if `this.checkServer(event_name, true)` is invoked (note the `true` parameter, which defaults to `false` if not provided). This component doesn't call it, instead calling `checkLocalStorage()`.

Here's the flow of this component, as I understand it:

- Constructor called
- Constructor invokes `checkPollStorage()` (line 29)
- In `checkPollStorage()` (line 219):
    - If there's nothing in local storage, indicate the component is ready (line 234).
    - Otherwise, load from the `getpollresults` endpoint (line 228)
        - On success, invoke `showPollResults()` (line 157); at the end of this function, indicate the component is ready (line 216)
        - On failure, indicate the component is ready (line 232).

...which is one of the more complex flows I've seen in the components.